### ISSUE TITLE
Resume training problem

### ISSUE THREAD

**Issue by fcoberrios** [keyth72/PedalNetRT#8]
I did a 4 hours training (got to 1056 epochs, 0.147 loss) but when i wanted to resume the training ive run into this issue 

H:\WAVENET\PedalNetRT-master>python train.py --resume_training="lightning_logs\version_0\checkpoints\epoch=1056.ckpt"
H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Unsupported `ReduceOp` for distributed computing.
  warnings.warn(*args, **kwargs)
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
CUDA_VISIBLE_DEVICES: [0]
H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Could not log computational graph since the `model.example_input_array` attribute is not set or `input_array` was not given
  warnings.warn(*args, **kwargs)

  | Name    | Type    | Params
------------------------------------
0 | wavenet | WaveNet | 18 K
Validation sanity check: 0it [00:00, ?it/s]H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Unsupported `ReduceOp` for distributed computing.
  warnings.warn(*args, **kwargs)
H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Unsupported `ReduceOp` for distributed computing.
  warnings.warn(*args, **kwargs)
H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Unsupported `ReduceOp` for distributed computing.
  warnings.warn(*args, **kwargs)
H:\Users\Francisco\anaconda3\lib\site-packages\pytorch_lightning\utilities\distributed.py:37: UserWarning: Unsupported `ReduceOp` for distributed computing.
  warnings.warn(*args, **kwargs)
Training: 0it [00:00, ?it/s]Saving latest checkpoint..
Training: 0it [00:00, ?it/s]

Link to the files

https://www.dropbox.com/sh/5ifigwcqyi3coqs/AACHB8gxs8khHSNP98rHddkAa?dl=0

**Comment by keyth72** [comment]
I got the same thing with your model, when you resume training are you using either a different computer, or switching from CPU to GPU, or GPU to CPU? If so, it might be related to training the same model on different devices, which can be done, but would need some code modifications to account for.

**Comment by fcoberrios** [comment]
> I got the same thing with your model, when you resume training are you using either a different computer, or switching from CPU to GPU, or GPU to CPU? If so, it might be related to training the same model on different devices, which can be done, but would need some code modifications to account for.

Mmm weird, im using the same computer and using the GPU, also, i tried to train the same model from scratch and pause it after less than 10 epochs (Ctrl+C) and when i try to resume it using 
--resume_training="lightning_logs\version_0\checkpoints\epoch=6.ckpt" (This is just a code example)
i dont get any errors.
Im training the model again from scratch and will pause it this time at 500 epochs, i will come back again if i can resume the model or not :)

**Comment by keyth72** [comment]
I think I got it, so the default max_epochs in PyTorch is 1000, so if it’s over 1000 already it thinks it’s finished training. I thought it would grab that number from the default args, but I’ll need to look into how to set that properly.

**Comment by fcoberrios** [comment]
> I think I got it, so the default max_epochs in PyTorch is 1000, so if it’s over 1000 already it thinks it’s finished training. I thought it would grab that number from the default args, but I’ll need to look into how to set that properly.

Yeah!, while i wasnt in my pc the training got to 1000 epochs and then finished, and then when i wanted to resume it (from the best loss of 0.14 or something, 993 epochs) it got to 1000 epochs and finished hahaha.

**Comment by keyth72** [comment]
Ok it's been updated, if you have any problems resuming training let me know.  You should also be able to update the max_epochs as well, if you get to the end and want to train further.  Also added a cpu flag to force cpu training.

**Comment by fcoberrios** [comment]
> Ok it's been updated, if you have any problems resuming training let me know. You should also be able to update the max_epochs as well, if you get to the end and want to train further. Also added a cpu flag to force cpu training.

Thanks!, im going to start training right away :), should have a good trained model for tomorrow, can you make a spreadsheet/facebook group to share models and data? :D

**Comment by keyth72** [comment]
I uploaded a spreadsheet, check out the google docs link under the "Analysis" section to log data. We'll see about a Facebook group, for now you can make a pull request to add any model's you'd like to share. 

I think you're still training on the GPU, as long as you don't provide the "--cpu=1" flag, it won't train on the cpu.  You will still have cpu usage though, just from running the program and when the cpu interacts with the gpu.  Might want to run it overnight if you need your pc for other things!

**Comment by fcoberrios** [comment]
Weird, i could only train to 1499 epochs, can you check your code please?
![image](https://user-images.githubusercontent.com/73447045/97355090-2f601b80-1875-11eb-822e-e0aa9335574b.png)

![image](https://user-images.githubusercontent.com/73447045/97355288-777f3e00-1875-11eb-81fd-7ff3ce87879f.png)

Also, im going to compare the old train.py to check if the cpu was used too, because if i can remember well it wasnt being used, only the gpu which was great :)
